src/shared/components/views/FeaturesView.tsx

```diff
@@ -28,33 +28,36 @@ export type ProcessedFeature = {
   featureId?: string;
   start: number;
   end: number;
-  startModifier: LocationModifier;
-  endModifier: LocationModifier;
+  startModifier?: LocationModifier;
+  endModifier?: LocationModifier;
   type: FeatureType;
   description?: string;
   evidences?: Evidence[];
   sequence?: string;
   locations?: { fragments: Fragment[] }[];
 };
 
-export type ColumnConfig = (
+export type ColumnConfig<FeatureType> = (
   callback: (
     evidenceData: EvidenceData,
     references: Evidence[] | undefined
   ) => void
-) => FeatureColumns;
+) => FeatureColumns<FeatureType>;
 
-type FeatureProps = {
+type FeatureProps<T> = {
+  features: T[];
+  columnConfig: ColumnConfig<T>;
+  // eslint-disable-next-line react/require-default-props
+  trackHeight?: number;
+  // eslint-disable-next-line react/require-default-props
   sequence?: string;
-  features: ProcessedFeature[];
-  columnConfig: ColumnConfig;
 };
 
-const FeaturesView: FC<FeatureProps> = ({
-  sequence,
-  features,
-  columnConfig,
-}): JSX.Element | null => {
+// ProcessedFeature | TransformedVariant | UniParcProcessedFeature
+const FeaturesView = <T extends Record<string, unknown>>(
```

Comment:
```suggestion
const FeaturesView = <T extends ProcessedFeature | TransformedVariant | UniParcProcessedFeature>(
```
I believe this might solve the below `as string` comment.

Reply:
🤦 of course